presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/PushPredicateIntoTableScan.java

```diff
@@ -131,7 +131,8 @@ private boolean arePlansSame(FilterNode filter, TableScanNode tableScan, PlanNod
 
         TableScanNode rewrittenTableScan = (TableScanNode) rewrittenFilter.getSource();
 
-        return Objects.equals(tableScan.getEnforcedConstraint(), rewrittenTableScan.getEnforcedConstraint());
+        return Objects.equals(tableScan.getEnforcedConstraint(), rewrittenTableScan.getEnforcedConstraint()) &&
+                Objects.equals(tableScan.getTable().getConnectorHandle(), rewrittenTableScan.getTable().getConnectorHandle());
```

Comment:
This should compare the table handles directly. This layer of the code (i.e., the optimizer) shouldn't be concerned about ConnectorTableHandle.